hax/treemakers/posrec.py

```diff
@@ -0,0 +1,269 @@
+import logging
+from hax.minitrees import TreeMaker
+from pax.PatternFitter import PatternFitter
+from pax.configuration import load_configuration
+from pax.InterpolatingMap import InterpolatingMap
+from pax import utils
+from pax import exceptions
+from pax.plugins.interaction_processing.S1AreaFractionTopProbability import binom_test as binom_test_pax
+from keras.models import model_from_json
+
+log = logging.getLogger('hax.posrec')
+
+
+class PositionReconstruction(TreeMaker):
+    """Stores position-reconstruction-related variables.
+
+    Provides:
+       - s1_pattern_fit_hax: S1 pattern likelihood computed with corrected
+                             position and areas
+       - s1_pattern_fit_hits_hax: S1 pattern likelihood computed with corrected
+                                  position and hits
+       - s1_pattern_fit_bottom_hax: S1 pattern likelihood computed with corrected
+                                    position and bottom array area
+       - s1_pattern_fit_bottom_hits_hax: S1 pattern likelihood computed with corrected
+                                         position and bottom array hits
+
+       - s1_area_fraction_top_probability_hax: S1 AFT computed with corrected position
+
+       - x_observed_nn_tf: TensorFlow NN reconstructed x position
+       - y_observed_nn_tf: TensorFlow NN reconstructed y position
+
+       - s1_area_upper_injection_fraction: s1 area fraction near Rn220 injection points (near PMT 131)
+       - s1_area_lower_injection_fraction: s1 area fraction near Rn220 injection points (near PMT 243)
+    """
+    __version__ = '0.16'
+    extra_branches = ['peaks.area_per_channel[260]',
+                      'peaks.hits_per_channel[260]',
+                      'peaks.n_saturated_per_channel[260]',
+                      'interactions.x', 'interactions.y', 'interactions.z']
+
+    extra_metadata = hax.config['corrections_definitions']
+
+    loaded_tfnn_model_json_name = None
+    nn_tensorflow = None
+
+    loaded_s1_aft_map_name = None
+    s1_aft_map = None
+
+    def get_correction(self, correction_name):
+        """Return the file to use for a correction or reconstruction algorithm"""
+
+        if ('corrections_definitions' not in hax.config) \
+                or (correction_name not in hax.config['corrections_definitions']):
+            return None
+
+        for entry in hax.config['corrections_definitions'][correction_name]:
+
+            if 'run_min' not in entry or self.run_number < entry['run_min']:
+                continue
+
+            if 'run_max' not in entry or self.run_number <= entry['run_max']:
+
+                # For corrections and other maps
+                if 'correction' in entry:
+                    return entry['correction']
+
+                # For NN files
+                elif 'weights' in entry and 'model' in entry:
+                    return entry['weights'], entry['model']
+
+        return None
+
+    def load_map(self, name, loaded_map, loaded_name):
```

Comment:
Seriously? Sorry this is now getting way too sloppy. This code is straight up copy-pasted out of a different treemaker.

Reply:
Fixed in https://github.com/XENON1T/hax/pull/185